packages/block-library/src/image/approvalDialog.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import {
+	ReactCompareSlider,
+	ReactCompareSliderImage,
+} from 'react-compare-slider'; // eslint-disable-line import/no-unresolved
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { createInterpolateElement, useState } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as uploadStore } from '@wordpress/upload-media';
+
+const numberFormatter = Intl.NumberFormat( 'en', {
+	notation: 'compact',
+	style: 'unit',
+	unit: 'byte',
+	unitDisplay: 'narrow',
+	// @ts-ignore
+	roundingPriority: 'lessPrecision',
+	maximumSignificantDigits: 2,
+	maximumFractionDigits: 2,
+} );
+
+const diffFormatter = Intl.NumberFormat( 'en', {
+	notation: 'compact',
+	style: 'percent',
+	maximumFractionDigits: 2,
+} );
+
+interface ApprovalDialogProps {
+	id: number;
+}
+
+export function ApprovalDialog( { id }: ApprovalDialogProps ) {
+	const { isPendingApproval, comparison } = useSelect(
+		( select ) => ( {
+			// This allows showing only one approval modal at a time if there
+			// are multiple pending items.
+			isPendingApproval: id
+				? select( uploadStore ).isFirstPendingApprovalByAttachmentId(
+						id
+				  )
+				: false,
+			comparison: id
+				? select( uploadStore ).getComparisonDataForApproval( id )
+				: null,
+		} ),
+		[ id ]
+	);
+
+	const { rejectApproval, grantApproval } = useDispatch( uploadStore );
+	const [ , setOpen ] = useState( false );
+	const closeModal = () => setOpen( false );
+	const onApprove = () => {
+		closeModal();
+		void grantApproval( id );
+	};
+
+	const onReject = () => {
+		closeModal();
+		void rejectApproval( id );
+	};
+
+	if ( ! isPendingApproval || ! comparison ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			title={ __( 'Compare media quality' ) }
+			onRequestClose={ onReject }
+		>
+			<div className="compression-comparison-modal__labels">
+				<p>
+					{ sprintf(
+						/* translators: %s: file size. */
+						__( 'Old version: %s' ),
+						numberFormatter.format( comparison.oldSize )
+					) }
+				</p>
+				<p>
+					{ sprintf(
+						/* translators: %s: file size. */
+						__( 'New version: %s' ),
+						numberFormatter.format( comparison.newSize )
+					) }
+				</p>
+			</div>
+			<p>
+				{ createInterpolateElement(
+					comparison.sizeDiff < 0
+						? sprintf(
+								/* translators: %s: file size decrease in percent. */
+								__( 'The new version is <b>%s smaller</b>!' ),
+								diffFormatter.format( comparison.sizeDiff )
+						  )
+						: sprintf(
+								/* translators: %s: file size increase in percent. */
+								__( 'The new version is <b>%s bigger</b> :(' ),
```

Comment:
I was testing the PR and this happened to me (33% bigger in 4MB image). When this can happen? It feels weird to click an `optimize` button and end up with a bigger file.

Can we handle this differently? At least in the level of wording - ex: see if can be optimized.

Reply:
That's impossible to say without knowing the image and the settings used. In this PR things like image quality or preferred file format aren't configurable yet either, so depending on the source image a size increase is definitely possible.

I'd suggest trying it again with https://github.com/swissspidy/media-experiments and opening an issue there if it happens, ideally uploading the unmodified original image somewhere too for inspection.